Frends.PDF.Create/Frends.PDF.Create.Tests/UnitTests.cs

```diff
@@ -0,0 +1,139 @@
+with some tab
+    one
+        two
+            three. Then end with scandic letter ö and russian word код.";
+
+            var result = CallCreatePdf(new PageContentElement[] { _header, _footer, _title, _paragraphContent, new PageContentElement { ContentType = ElementType.PageBreak }, _tableContent });
+
+            Assert.IsTrue(File.Exists(_destinationFullPath));
+            Assert.IsTrue(result.Success);
+        }
+
+        [Test]
+        public void WritePdf_DoesNotFailIfContentIsEmpty()
+        {
+            _paragraphContent.Text = null;
+            var result = CallCreatePdf(new PageContentElement[] { _paragraphContent });
+
+            Assert.IsTrue(File.Exists(_destinationFullPath));
+            Assert.IsTrue(result.Success);
+        }
+
+        [Test]
+        public void WritePdf_ThrowsExceptionIfFileExists()
+        {
+            _options.ThrowErrorOnFailure = true;
+            _fileProperties.FileExistsAction = FileExistsActionEnum.Error;
+
+            // Run once so file exists.
+            CallCreatePdf(new PageContentElement[] { _paragraphContent });
+
+            var result = Assert.Throws<Exception>(() => CallCreatePdf(new PageContentElement[] { _paragraphContent }));
+        }
+
+        [Test]
+        public void WritePdf_RenamesFilesIfAlreadyExists()
+        {
+            _fileProperties.FileExistsAction = FileExistsActionEnum.Rename;
+
+            // Create 3 files.
+            var result1 = CallCreatePdf(new PageContentElement[] { _paragraphContent });
+            var result2 = CallCreatePdf(new PageContentElement[] { _paragraphContent });
+            var result3 = CallCreatePdf(new PageContentElement[] { _paragraphContent });
+
+            Assert.IsTrue(File.Exists(result1.FileName));
+            Assert.AreEqual(_destinationFullPath, result1.FileName);
+            Assert.IsTrue(File.Exists(result2.FileName));
+            Assert.IsTrue(result2.FileName.Contains("_(1)"));
+            Assert.IsTrue(File.Exists(result3.FileName));
+            Assert.IsTrue(result3.FileName.Contains("_(2)"));
+        }
+
+        [Test]
+        public void WritePdf_LogoNotFound()
+        {
+            _fileProperties.FileExistsAction = FileExistsActionEnum.Overwrite;
+
+            var logoPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"Files\no_such_logo.png");
+            var header = new PageContentElement { ContentType = ElementType.Header, FontFamily = "Times New Roman", FontSize = 8, FontStyle = FontStyleEnum.Regular, LineSpacingInPt = 11, ParagraphAlignment = ParagraphAlignmentEnum.Right, SpacingAfterInPt = 0, SpacingBeforeInPt = 8, ImagePath = logoPath, HeaderFooterStyle = HeaderFooterStyleEnum.LogoText, BorderWidthInPt = 0.5, ImageHeightInCm = 0.5 };
+            header.Text = @"This is a header";
+
+            var result = Assert.Throws<FileNotFoundException>(() => CallCreatePdf(new PageContentElement[] { header }));

```

Comment:
Do we have a change to wrap the actual exception from creating the PDF? It would be nice to have e.g. this FileNotFoundException as inner exception within our wrapper exception. And the wrapping exception could say something more humane like "An error occurred while writing PDF". This way e.g. on process instance list you would be able to tell what tast this error is coming from and we would still retain the original error in inner exception. This actually is valid for every other exception that is tested here.